qiskit/chemistry/drivers/fcidumpd/parser.py

```diff
@@ -0,0 +1,236 @@
+    metadata = fcidump_str[:namelist_end.start(0)]
+    metadata = ' '.join(metadata.split())  # replace duplicate whitespace and newlines
+    # we know what elements to look for so we don't get too fancy with the parsing
+    # pattern explanation:
+    #  .*?      any text
+    #  (*|*),   match either part of this group followed by a comma
+    #  [-+]?    match up to a single - or +
+    #  \d*.\d+  number format
+    pattern = r'.*?([-+]?\d*\.\d+|[-+]?\d+),'
+    # we parse the values in the order in which they are listed in Knowles1989
+    _norb = re.search('NORB'+pattern, metadata)
+    norb = int(_norb.groups()[0])
+    output['NORB'] = norb
+    _nelec = re.search('NELEC'+pattern, metadata)
+    output['NELEC'] = int(_nelec.groups()[0])
+    # the rest of these values may occur and are set to their defaults otherwise
+    _ms2 = re.search('MS2'+pattern, metadata)
+    output['MS2'] = int(_ms2.groups()[0]) if _ms2 else 0
+    _isym = re.search('ISYM'+pattern, metadata)
+    output['ISYM'] = int(_isym.groups()[0]) if _isym else 1
+    # ORBSYM holds a list, thus it requires a little different treatment
+    _orbsym = re.search(r'ORBSYM.*?'+r'(\d+),'*norb, metadata)
+    output['ORBSYM'] = [int(s) for s in _orbsym.groups()] if _orbsym else [1] * len(norb)
+    _iprtim = re.search('IPRTIM'+pattern, metadata)
+    output['IPRTIM'] = int(_iprtim.groups()[0]) if _iprtim else -1
+    _int = re.search('INT'+pattern, metadata)
+    output['INT'] = int(_int.groups()[0]) if _int else 5
+    _memory = re.search('MEMORY'+pattern, metadata)
+    output['MEMORY'] = int(_memory.groups()[0]) if _memory else 10000
+    _core = re.search('CORE'+pattern, metadata)
+    output['CORE'] = float(_core.groups()[0]) if _core else 0.0
+    _maxit = re.search('MAXIT'+pattern, metadata)
+    output['MAXIT'] = int(_maxit.groups()[0]) if _maxit else 25
+    _thr = re.search('THR'+pattern, metadata)
+    output['THR'] = float(_thr.groups()[0]) if _thr else 1e-5
+    _thrres = re.search('THRRES'+pattern, metadata)
+    output['THRRES'] = float(_thrres.groups()[0]) if _thrres else 0.1
+    _nroot = re.search('NROOT'+pattern, metadata)
+    output['NROOT'] = int(_nroot.groups()[0]) if _nroot else 1
+
+    # If the FCIDump file resulted from an unrestricted spin calculation the indices will label spin
+    # rather than molecular orbitals. This means, that a line must exist which encodes the
+    # coefficient for the spin orbital with index (norb*2, norb*2). By checking for such a line we
+    # can distinguish between unrestricted and restricted FCIDump files.
+    _uhf = bool(re.search(r'.*(\s+{}\s+{}\s+0\s+0)'.format(norb*2, norb*2),
+                          fcidump_str[namelist_end.start(0):]))
+
+    # the rest of the FCIDump will hold lines of the form x i a j b
+    # a few cases have to be treated differently:
+    # i, a, j and b are all zero: x is the core energy
+    # TODO: a, j and b are all zero: x is the energy of the i-th MO  (often not supported)
```

Comment:
Is this TODO still open? Should it be added or is that for later?

Reply:
Same as the other TODOs: these are for later.
See also the [comment above](https://github.com/Qiskit/qiskit-aqua/pull/859#issuecomment-600604937)